### PR TITLE
Lesson 1 5

### PR DESCRIPTION
**Pull request by Uni-syeda**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,18 +1,32 @@
 import reactLogo from "./assets/react.svg";
 import viteLogo from "/vite.svg";
 import "./App.css";
-import { useState } from "react"; 
-import TodoList from "./TodoList";
+import { useState, useEffect } from "react";
+import TodoList from "./TodoList.jsx";
```
Comment (nicholasgalante1997) [inline src/App.jsx:5]: You shouldn't need this .jsx extension since you're using a bundler (In this case vite) 👍

File: src/App.jsx [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
 import AddTodoForm from "./AddTodoForm";
 
+function useSemiPersistentState() {
+  const existingTodo = JSON.parse(localStorage.getItem("savedTodoList")) ?? [];
```
Comment (nicholasgalante1997) [inline src/App.jsx:9]: Nice use of ?? here

For more info on the nullish coalescing operator, you can read this mdn article

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

**Review by nicholasgalante1997 — APPROVED** [review]

**Comment by nicholasgalante1997** [comment]
Great job! You did great with this assignment. Go ahead and merge this PR whenever you feel ready 👍

File: src/App.jsx (hunk 1 above)
Comment (Uni-syeda) [reply src/App.jsx:5]: got it! thank you :)

**Pull request merged**